gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,233 @@
+
+    public List<DiagnosticReport> mapDiagnosticReports(RCMRMT030101UK04EhrExtract ehrExtract, Patient patient,
+        List<Encounter> encounters, String practiceCode) {
+        var compositions = getCompositionsContainingClusterCompoundStatement(ehrExtract);
+        return compositions.stream()
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .flatMap(CompoundStatementUtil::extractAllCompoundStatements)
+            .filter(Objects::nonNull)
+            .filter(this::isDiagnosticReportCandidate)
+            .map(compoundStatement -> {
+                DiagnosticReport diagnosticReport = createDiagnosticReport(
+                    compoundStatement, patient, compositions, encounters, practiceCode
+                );
+                getIssued(ehrExtract, compositions, compoundStatement).ifPresent(diagnosticReport::setIssuedElement);
+                return diagnosticReport;
+            }).toList();
+    }
+
+    private DiagnosticReport createDiagnosticReport(RCMRMT030101UK04CompoundStatement compoundStatement, Patient patient,
+        List<RCMRMT030101UK04EhrComposition> compositions, List<Encounter> encounters, String practiceCode) {
+        final DiagnosticReport diagnosticReport = new DiagnosticReport();
+        final String id = compoundStatement.getId().get(0).getRoot();
+
+        diagnosticReport.setMeta(generateMeta(META_PROFILE_URL_SUFFIX));
+        diagnosticReport.setId(id);
+        diagnosticReport.addIdentifier(buildIdentifier(id, practiceCode));
+        diagnosticReport.setCode(createCodeableConcept(compoundStatement));
+        diagnosticReport.setSubject(new Reference(patient));
+        diagnosticReport.setSpecimen(getSpecimenReferences(compoundStatement));
+        createIdentifierExtension(compoundStatement.getId()).ifPresent(diagnosticReport::addIdentifier);
+        buildContext(compositions, encounters, compoundStatement).ifPresent(diagnosticReport::setContext);
+        setResultReferences(compoundStatement, diagnosticReport);
+
+        return diagnosticReport;
+    }
+
+    public void mapChildObservationComments(RCMRMT030101UK04EhrExtract ehrExtract, List<Observation> observationComments) {
+        getCompositionsContainingClusterCompoundStatement(ehrExtract)
+            .stream()
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .map(RCMRMT030101UK04Component4::getCompoundStatement)
+            .filter(Objects::nonNull)
+            .flatMap(compoundStatement -> compoundStatement.getComponent().stream())
+            .map(RCMRMT030101UK04Component02::getNarrativeStatement)
+            .filter(Objects::nonNull)
+            .map(narrativeStatement -> getObservationCommentById(observationComments, narrativeStatement.getId().getRoot()))
+            .flatMap(Optional::stream)
+            .forEach(observationComment -> {
+                observationComment.setEffective(null);
```

Comment:
Does setting a null value make sense?

Reply:
It's about removing the value, I've set it to be null so there is no empty field mapped in the final JSON